commit e768ae9db49404408d13d22a4e9f7d81b095dab8
Author: Devlin Cronin <rdevlin.cronin@chromium.org>
Date:   Fri May 22 16:01:04 2026 -0700

    [Extensions] Remove kExtensionManifestV2Disabled feature
    
    The kExtensionManifestV2Disabled feature has been default-enabled for
    over a year. Remove the feature and the effectively-dead code.
    
    Any tests that relied on being in the "warning" phase (i.e., with the
    kExtensionManifestV2Disabled) for their sole behavior testing are
    removed, since this stage is no longer reachable.
    
    This CL should have no behavior change in production.
    
    Bug: 431097630
    Change-Id: I2676cbcea52e4ca4436f3d7060c8a736c7961190
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7813942
    Reviewed-by: Tim <tjudkins@chromium.org>
    Commit-Queue: Devlin Cronin <rdevlin.cronin@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#1635281}

diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
index ef42ab74329e2..6bead036370b1 100644
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -8858,12 +8858,6 @@ const FeatureEntry kFeatureEntries[] = {
      FEATURE_VALUE_TYPE(
          feature_engagement::kIPHExtensionsRequestAccessButtonFeature)},
 
-    {"extension-manifest-v2-deprecation-disabled",
-     flag_descriptions::kExtensionManifestV2DeprecationDisabledName,
-     flag_descriptions::kExtensionManifestV2DeprecationDisabledDescription,
-     kOsDesktop,
-     FEATURE_VALUE_TYPE(extensions_features::kExtensionManifestV2Disabled)},
-
     {"extension-manifest-v2-deprecation-unsupported",
      flag_descriptions::kExtensionManifestV2DeprecationUnsupportedName,
      flag_descriptions::kExtensionManifestV2DeprecationUnsupportedDescription,
diff --git a/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc b/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc
index 7cd4e74a893ef..48d97605d5e7b 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc
+++ b/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc
@@ -3415,110 +3415,20 @@ TEST_F(DeveloperPrivateApiSupervisedUserUnitTest,
     EXPECT_THAT(error, testing::HasSubstr("Child account"));
 }
 
-// Test suite for cases where the user is in the  MV2 deprecation "warning"
-// experiment phase.
-class DeveloperPrivateApiWithMV2DeprecationWarningUnitTest
-    : public DeveloperPrivateApiUnitTest {
- public:
-  DeveloperPrivateApiWithMV2DeprecationWarningUnitTest() {
-    feature_list_.InitWithFeatures(
-        /*enabled_features=*/{},
-        /*disabled_features=*/{
-            extensions_features::kExtensionManifestV2Disabled,
-            extensions_features::kExtensionManifestV2Unsupported});
-  }
-
- private:
-  base::test::ScopedFeatureList feature_list_;
-};
-
 // Test suite for cases where the user is in the  MV2 deprecation "disabled"
 // experiment phase.
 class DeveloperPrivateApiWithMV2DeprecationDisabledUnitTest
     : public DeveloperPrivateApiUnitTest {
  public:
   DeveloperPrivateApiWithMV2DeprecationDisabledUnitTest() {
-    feature_list_.InitWithFeatures(
-        {extensions_features::kExtensionManifestV2Disabled},
-        {extensions_features::kExtensionManifestV2Unsupported});
+    feature_list_.InitAndDisableFeature(
+        extensions_features::kExtensionManifestV2Unsupported);
   }
 
  private:
   base::test::ScopedFeatureList feature_list_;
 };
 
-// Extension of manifest version 2 is not supported on Android.
-#if !BUILDFLAG(IS_ANDROID)
-TEST_F(DeveloperPrivateApiWithMV2DeprecationWarningUnitTest,
-       TestAcknowledgingAnExtension) {
-  // Add an extension that is affected by the MV2 deprecation.
-  scoped_refptr<const Extension> extension =
-      ExtensionBuilder("ext").SetManifestVersion(2).Build();
-  registrar()->AddExtension(extension.get());
-
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-
-  base::ListValue args;
-  args.Append(extension->id());
-
-  // Dismiss the extension's notice.
-  auto dismiss_notice_function = base::MakeRefCounted<
-      api::DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction>();
-  dismiss_notice_function->set_source_context_type(mojom::ContextType::kWebUi);
-  EXPECT_TRUE(RunFunction(dismiss_notice_function, args));
-
-  // Extension's notice should be marked as acknowledged.
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_TRUE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-}
-
-TEST_F(DeveloperPrivateApiWithMV2DeprecationWarningUnitTest,
-       TestAcknowledgingANonAffectedExtension) {
-  // Add an extension that is not affected by the MV2 deprecation.
-  scoped_refptr<const Extension> extension = ExtensionBuilder("ext").Build();
-  registrar()->AddExtension(extension.get());
-
-  std::string args = base::StringPrintf(R"(["%s"])", extension->id().c_str());
-  auto dismiss_notice_function = base::MakeRefCounted<
-      api::DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction>();
-  dismiss_notice_function->set_source_context_type(mojom::ContextType::kWebUi);
-
-  // Cannot dismiss an extension's notice whe the extension is not affected by
-  // the MV2 deprecation.
-  std::string error = api_test_utils::RunFunctionAndReturnError(
-      dismiss_notice_function, args, profile());
-  EXPECT_EQ(error,
-            ErrorUtils::FormatErrorMessage(
-                "Extension with ID '*' is not affected by the MV2 deprecation.",
-                extension->id()));
-
-  // Extension notice should not be marked as acknowledged.
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-}
-#endif  // !BUILDFLAG(IS_ANDROID)
-
-TEST_F(DeveloperPrivateApiWithMV2DeprecationWarningUnitTest,
-       TestAcknowledgingNoticeGlobally) {
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNoticeGlobally());
-
-  auto update_profile_function = base::MakeRefCounted<
-      api::DeveloperPrivateUpdateProfileConfigurationFunction>();
-  update_profile_function->set_source_context_type(mojom::ContextType::kWebUi);
-
-  base::ListValue args;
-  args.Append(base::DictValue().Set("isMv2DeprecationNoticeDismissed", true));
-  EXPECT_TRUE(RunFunction(update_profile_function, args));
-
-  EXPECT_TRUE(experiment_manager->DidUserAcknowledgeNoticeGlobally());
-}
-
 // Extension of manifest version 2 is not supported on Android.
 #if !BUILDFLAG(IS_ANDROID)
 TEST_F(DeveloperPrivateApiWithMV2DeprecationDisabledUnitTest,
diff --git a/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc b/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc
index 3d908a7ea237f..3b7c817bfcea5 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc
+++ b/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc
@@ -547,23 +547,13 @@ class DeveloperPrivateApiWithMV2DeprecationApiTest
     std::vector<base::test::FeatureRef> enabled_features;
     std::vector<base::test::FeatureRef> disabled_features;
     switch (experiment_stage_) {
-      case MV2ExperimentStage::kWarning:
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-        break;
       case MV2ExperimentStage::kDisableWithReEnable:
-        enabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         disabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
         break;
       case MV2ExperimentStage::kUnsupported:
         enabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
     }
 
     feature_list_.InitWithFeatures(enabled_features, disabled_features);
@@ -581,13 +571,10 @@ class DeveloperPrivateApiWithMV2DeprecationApiTest
 INSTANTIATE_TEST_SUITE_P(
     ,
     DeveloperPrivateApiWithMV2DeprecationApiTest,
-    testing::Values(MV2ExperimentStage::kWarning,
-                    MV2ExperimentStage::kDisableWithReEnable,
+    testing::Values(MV2ExperimentStage::kDisableWithReEnable,
                     MV2ExperimentStage::kUnsupported),
     [](const testing::TestParamInfo<MV2ExperimentStage>& info) {
       switch (info.param) {
-        case MV2ExperimentStage::kWarning:
-          return "WarningExperiment";
         case MV2ExperimentStage::kDisableWithReEnable:
           return "DisableExperiment";
         case MV2ExperimentStage::kUnsupported:
@@ -626,16 +613,6 @@ IN_PROC_BROWSER_TEST_P(DeveloperPrivateApiWithMV2DeprecationApiTest,
   std::string args = base::StringPrintf(R"(["%s"])", extension->id().c_str());
 
   switch (experiment_stage()) {
-    case MV2ExperimentStage::kWarning:
-      api_test_utils::RunFunction(dismiss_notice_function.get(), args,
-                                  profile());
-
-      // Extension's notice should be marked as acknowledged.
-      EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-      EXPECT_TRUE(
-          experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-      break;
-
     case MV2ExperimentStage::kDisableWithReEnable: {
       // The function will trigger a dialog for this stage. Add a waiter for the
       // dialog.
diff --git a/chrome/browser/extensions/api/developer_private/developer_private_functions.cc b/chrome/browser/extensions/api/developer_private/developer_private_functions.cc
index 17baf6668e246..221a73b5fd8ec 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_functions.cc
+++ b/chrome/browser/extensions/api/developer_private/developer_private_functions.cc
@@ -2582,12 +2582,6 @@ DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::Run() {
   MV2ExperimentStage experiment_stage =
       experiment_manager->GetCurrentExperimentStage();
   switch (experiment_stage) {
-    case MV2ExperimentStage::kWarning: {
-      // Immediately dismiss the notice.
-      DismissExtensionNotice();
-      return RespondNow(NoArguments());
-    }
-
     case MV2ExperimentStage::kDisableWithReEnable: {
       // Prompt for user confirmation before dismissing the notice.
       if (accept_bubble_for_testing_.has_value()) {
diff --git a/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc b/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc
index b0527b70729e2..fc297c3be479f 100644
--- a/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc
+++ b/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc
@@ -1337,23 +1337,13 @@ class ExtensionInfoGeneratorWithMV2DeprecationUnitTest
 
     experiment_stage_ = GetParam();
     switch (experiment_stage_) {
-      case MV2ExperimentStage::kWarning:
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-        break;
       case MV2ExperimentStage::kDisableWithReEnable:
-        enabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         disabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
         break;
       case MV2ExperimentStage::kUnsupported:
         enabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         break;
     }
 
@@ -1373,13 +1363,10 @@ class ExtensionInfoGeneratorWithMV2DeprecationUnitTest
 INSTANTIATE_TEST_SUITE_P(
     All,
     ExtensionInfoGeneratorWithMV2DeprecationUnitTest,
-    testing::Values(MV2ExperimentStage::kWarning,
-                    MV2ExperimentStage::kDisableWithReEnable,
+    testing::Values(MV2ExperimentStage::kDisableWithReEnable,
                     MV2ExperimentStage::kUnsupported),
     [](const testing::TestParamInfo<MV2ExperimentStage>& info) {
       switch (info.param) {
-        case MV2ExperimentStage::kWarning:
-          return "WarningExperiment";
         case MV2ExperimentStage::kDisableWithReEnable:
           return "DisableExperiment";
         case MV2ExperimentStage::kUnsupported:
diff --git a/chrome/browser/extensions/api/management/management_api_unittest.cc b/chrome/browser/extensions/api/management/management_api_unittest.cc
index 0e398b187e1a7..f7b345c2cb20c 100644
--- a/chrome/browser/extensions/api/management/management_api_unittest.cc
+++ b/chrome/browser/extensions/api/management/management_api_unittest.cc
@@ -891,9 +891,8 @@ class ManagementApiUnitTestMV2DisableWithReEnableUnitTest
     : public ManagementApiUnitTest {
  public:
   ManagementApiUnitTestMV2DisableWithReEnableUnitTest() {
-    feature_list_.InitWithFeatures(
-        {extensions_features::kExtensionManifestV2Disabled},
-        {extensions_features::kExtensionManifestV2Unsupported});
+    feature_list_.InitAndDisableFeature(
+        extensions_features::kExtensionManifestV2Unsupported);
   }
   ~ManagementApiUnitTestMV2DisableWithReEnableUnitTest() override = default;
 
diff --git a/chrome/browser/extensions/api/webstore_private/extension_install_status_unittest.cc b/chrome/browser/extensions/api/webstore_private/extension_install_status_unittest.cc
index f449ed087c0a4..54d3ab2794077 100644
--- a/chrome/browser/extensions/api/webstore_private/extension_install_status_unittest.cc
+++ b/chrome/browser/extensions/api/webstore_private/extension_install_status_unittest.cc
@@ -767,85 +767,6 @@ TEST_F(ExtensionInstallStatusTestWithCloudPolicyChecks,
 }
 #endif  // BUILDFLAG(ENABLE_EXTENSIONS)
 
-// These tests have dependencies on ManifestV2ExperimentManager which is not
-// supported on Android (which only supports manifest V3).
-class ExtensionInstallStatusTestWithoutMv2Deprecation
-    : public ExtensionInstallStatusTest {
- public:
-  ExtensionInstallStatusTestWithoutMv2Deprecation() {
-    // This test assumes MV2 is not blocked by Chrome. Versions with MV2
-    // blocked by Chrome are exercised in
-    // `ExtensionInstallStatusTestWithMV2Deprecation`.
-    feature_list_.InitWithFeatures(
-        {}, {extensions_features::kExtensionManifestV2Disabled,
-             extensions_features::kExtensionManifestV2Unsupported});
-  }
-  ExtensionInstallStatusTestWithoutMv2Deprecation(
-      const ExtensionInstallStatusTestWithoutMv2Deprecation&) = delete;
-  ExtensionInstallStatusTestWithoutMv2Deprecation& operator=(
-      const ExtensionInstallStatusTestWithoutMv2Deprecation&) = delete;
-  ~ExtensionInstallStatusTestWithoutMv2Deprecation() override = default;
-
- private:
-  base::test::ScopedFeatureList feature_list_;
-};
-
-TEST_F(ExtensionInstallStatusTestWithoutMv2Deprecation,
-       ManifestVersionIsBlocked) {
-  EXPECT_EQ(
-      ExtensionInstallStatus::kInstallable,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/2));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kInstallable,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/3));
-  SetPolicy(pref_names::kManifestV2Availability,
-            std::make_unique<base::Value>(static_cast<int>(
-                internal::GlobalSettings::ManifestV2Setting::kDisabled)));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kBlockedByPolicy,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/2));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kInstallable,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/3));
-}
-
-TEST_F(ExtensionInstallStatusTestWithoutMv2Deprecation,
-       ManifestVersionIsBlockedWithExtensionRequest) {
-  SetPolicy(enterprise_reporting::kCloudExtensionRequestEnabled,
-            std::make_unique<base::Value>(true));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kCanRequest,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/2));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kCanRequest,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/3));
-  SetPolicy(pref_names::kManifestV2Availability,
-            std::make_unique<base::Value>(static_cast<int>(
-                internal::GlobalSettings::ManifestV2Setting::kDisabled)));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kBlockedByPolicy,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/2));
-  EXPECT_EQ(
-      ExtensionInstallStatus::kCanRequest,
-      GetInstallStatusSynchronously(kExtensionId, profile(), base::Version(),
-                                    Manifest::Type::TYPE_EXTENSION,
-                                    PermissionSet(), /*manifest_version=*/3));
-}
-
 // If an existing, installed extension is disabled due to corruption, then
 // GetWebstoreExtensionInstallStatus() should return kCorrupted.
 TEST_F(ExtensionInstallStatusTest, ExtensionCorrupted) {
@@ -928,21 +849,11 @@ TEST_F(SupervisedUserExtensionInstallStatusTest,
 #if BUILDFLAG(ENABLE_EXTENSIONS)
 // A test suite to toggle the behavior of the MV2 deprecation experiment.
 class ExtensionInstallStatusTestWithMV2Deprecation
-    : public ExtensionInstallStatusTest,
-      public testing::WithParamInterface<bool> {
+    : public ExtensionInstallStatusTest {
  public:
   ExtensionInstallStatusTestWithMV2Deprecation() {
-    std::vector<base::test::FeatureRef> enabled_features;
-    std::vector<base::test::FeatureRef> disabled_features(
-        {extensions_features::kExtensionManifestV2Unsupported});
-    if (GetParam()) {
-      enabled_features.push_back(
-          extensions_features::kExtensionManifestV2Disabled);
-    } else {
-      disabled_features.push_back(
-          extensions_features::kExtensionManifestV2Disabled);
-    }
-    feature_list_.InitWithFeatures(enabled_features, disabled_features);
+    feature_list_.InitAndDisableFeature(
+        extensions_features::kExtensionManifestV2Unsupported);
   }
   ~ExtensionInstallStatusTestWithMV2Deprecation() override = default;
 
@@ -950,19 +861,9 @@ class ExtensionInstallStatusTestWithMV2Deprecation
   base::test::ScopedFeatureList feature_list_;
 };
 
-using ExtensionInstallStatusTestWithMV2DeprecationEnabled =
-    ExtensionInstallStatusTestWithMV2Deprecation;
-
-INSTANTIATE_TEST_SUITE_P(All,
-                         ExtensionInstallStatusTestWithMV2Deprecation,
-                         testing::Bool());
-INSTANTIATE_TEST_SUITE_P(All,
-                         ExtensionInstallStatusTestWithMV2DeprecationEnabled,
-                         testing::Values(true));
-
 // Tests the webstore properly checks whether an extension can be installed
 // inline with the MV2 Deprecation experiments.
-TEST_P(ExtensionInstallStatusTestWithMV2Deprecation,
+TEST_F(ExtensionInstallStatusTestWithMV2Deprecation,
        MV2ExtensionsAreBlockedWithExperiment) {
   const ExtensionId kTestId(32, 'a');
 
@@ -976,8 +877,7 @@ TEST_P(ExtensionInstallStatusTestWithMV2Deprecation,
   // MV2 extensions should be unavailable if and only if the experiment is
   // enabled.
   ExtensionInstallStatus expected_status =
-      GetParam() ? ExtensionInstallStatus::kDeprecatedManifestVersion
-                 : ExtensionInstallStatus::kInstallable;
+      ExtensionInstallStatus::kDeprecatedManifestVersion;
   EXPECT_EQ(expected_status, GetInstallStatusSynchronously(
                                  kTestId, profile(), base::Version(),
                                  Manifest::TYPE_EXTENSION, PermissionSet(),
@@ -986,7 +886,7 @@ TEST_P(ExtensionInstallStatusTestWithMV2Deprecation,
 
 // An extension explicitly blocked by the admin should be considered blocked
 // by policy, rather than a deprecated manifest version.
-TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
+TEST_F(ExtensionInstallStatusTestWithMV2Deprecation,
        IdBlockedByPolicyTakesPriorityOverDeprecatedManifestVersion) {
   SetExtensionSettings(kExtensionSettingsWithIdBlocked);
   EXPECT_EQ(
@@ -998,7 +898,7 @@ TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
 
 // If an admin blocks all MV2 extensions, they should be considered blocked by
 // policy, rather than a deprecated manifest version.
-TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
+TEST_F(ExtensionInstallStatusTestWithMV2Deprecation,
        ManifestV2PolicyTakesPriorityOverDeprecatedManifestVersion) {
   SetPolicy(pref_names::kManifestV2Availability,
             std::make_unique<base::Value>(static_cast<int>(
@@ -1013,7 +913,7 @@ TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
 // Extensions that are installed and enabled should indicate such, even if they
 // are using a deprecated manifest version (since they are either re-enabled by
 // the user or are allowed by the admin).
-TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
+TEST_F(ExtensionInstallStatusTestWithMV2Deprecation,
        EnabledTakesPriorityOverDeprecatedManifestVersion) {
   ExtensionRegistry::Get(profile())->AddEnabled(CreateExtension(kExtensionId));
   EXPECT_EQ(
@@ -1026,7 +926,7 @@ TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
 // Extensions that are installed and disabled and have a deprecated manifest
 // version should indicate they are unsupported due to the manifest version.
 // Note, this applies even if they are disabled due to other reasons.
-TEST_P(ExtensionInstallStatusTestWithMV2DeprecationEnabled,
+TEST_F(ExtensionInstallStatusTestWithMV2Deprecation,
        DeprecatedManifestVersionTakesPriorityOverDisabled) {
   ExtensionRegistry::Get(profile())->AddDisabled(CreateExtension(kExtensionId));
   EXPECT_EQ(
diff --git a/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc b/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc
index ce248593903de..173a691f39916 100644
--- a/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc
+++ b/chrome/browser/extensions/api/webstore_private/webstore_private_api.cc
@@ -1611,9 +1611,6 @@ WebstorePrivateGetMV2DeprecationStatusFunction::Run() {
   api::webstore_private::MV2DeprecationStatus api_status =
       api::webstore_private::MV2DeprecationStatus::kInactive;
   switch (current_stage) {
-    case MV2ExperimentStage::kWarning:
-      api_status = api::webstore_private::MV2DeprecationStatus::kWarning;
-      break;
     case MV2ExperimentStage::kDisableWithReEnable:
       api_status = api::webstore_private::MV2DeprecationStatus::kSoftDisable;
       break;
diff --git a/chrome/browser/extensions/api/webstore_private/webstore_private_unittest.cc b/chrome/browser/extensions/api/webstore_private/webstore_private_unittest.cc
index f43d276f1b103..5b09fb2098324 100644
--- a/chrome/browser/extensions/api/webstore_private/webstore_private_unittest.cc
+++ b/chrome/browser/extensions/api/webstore_private/webstore_private_unittest.cc
@@ -886,23 +886,13 @@ WebstorePrivateManifestV2DeprecationUnitTest::
   std::vector<base::test::FeatureRef> enabled_features;
   std::vector<base::test::FeatureRef> disabled_features;
   switch (GetParam()) {
-    case MV2ExperimentStage::kWarning:
-      disabled_features.push_back(
-          extensions_features::kExtensionManifestV2Disabled);
-      disabled_features.push_back(
-          extensions_features::kExtensionManifestV2Unsupported);
-      break;
     case MV2ExperimentStage::kDisableWithReEnable:
-      enabled_features.push_back(
-          extensions_features::kExtensionManifestV2Disabled);
       disabled_features.push_back(
           extensions_features::kExtensionManifestV2Unsupported);
       break;
     case MV2ExperimentStage::kUnsupported:
       enabled_features.push_back(
           extensions_features::kExtensionManifestV2Unsupported);
-      disabled_features.push_back(
-          extensions_features::kExtensionManifestV2Disabled);
       break;
   }
 
@@ -912,13 +902,10 @@ WebstorePrivateManifestV2DeprecationUnitTest::
 INSTANTIATE_TEST_SUITE_P(
     ,
     WebstorePrivateManifestV2DeprecationUnitTest,
-    testing::Values(MV2ExperimentStage::kWarning,
-                    MV2ExperimentStage::kDisableWithReEnable,
+    testing::Values(MV2ExperimentStage::kDisableWithReEnable,
                     MV2ExperimentStage::kUnsupported),
     [](const testing::TestParamInfo<MV2ExperimentStage>& info) {
       switch (info.param) {
-        case MV2ExperimentStage::kWarning:
-          return "WarningExperiment";
         case MV2ExperimentStage::kDisableWithReEnable:
           return "DisableExperiment";
         case MV2ExperimentStage::kUnsupported:
@@ -938,9 +925,6 @@ TEST_P(WebstorePrivateManifestV2DeprecationUnitTest,
 
   std::string expected;
   switch (GetParam()) {
-    case MV2ExperimentStage::kWarning:
-      expected = "warning";
-      break;
     case MV2ExperimentStage::kDisableWithReEnable:
       expected = "soft_disable";
       break;
diff --git a/chrome/browser/extensions/manifest_v2_experiment_manager_browsertest.cc b/chrome/browser/extensions/manifest_v2_experiment_manager_browsertest.cc
index 58a415777773f..918b8712a15e8 100644
--- a/chrome/browser/extensions/manifest_v2_experiment_manager_browsertest.cc
+++ b/chrome/browser/extensions/manifest_v2_experiment_manager_browsertest.cc
@@ -4,6 +4,8 @@
 
 #include "extensions/browser/manifest_v2_experiment_manager.h"
 
+#include <algorithm>
+
 #include "base/one_shot_event.h"
 #include "base/strings/stringprintf.h"
 #include "base/test/bind.h"
@@ -55,28 +57,23 @@ MV2ExperimentStage GetExperimentStageForTest(std::string_view test_name) {
     const char* test_name;
     MV2ExperimentStage stage;
   } test_stages[] = {
-      {"PRE_PRE_ExtensionsAreDisabledOnStartup", MV2ExperimentStage::kWarning},
-      {"PRE_ExtensionsAreDisabledOnStartup", MV2ExperimentStage::kWarning},
+      {"PRE_ExtensionsAreDisabledOnStartup",
+       MV2ExperimentStage::kDisableWithReEnable},
       {"ExtensionsAreDisabledOnStartup",
        MV2ExperimentStage::kDisableWithReEnable},
-      {"PRE_PRE_ExtensionsCanBeReEnabledByUsers", MV2ExperimentStage::kWarning},
+      {"PRE_PRE_ExtensionsCanBeReEnabledByUsers",
+       MV2ExperimentStage::kDisableWithReEnable},
       {"PRE_ExtensionsCanBeReEnabledByUsers",
        MV2ExperimentStage::kDisableWithReEnable},
       {"ExtensionsCanBeReEnabledByUsers",
        MV2ExperimentStage::kDisableWithReEnable},
       {"ExtensionsAreReEnabledWhenUpdatedToMV3",
        MV2ExperimentStage::kDisableWithReEnable},
-      {"PRE_MarkingNoticeAsAcknowledged", MV2ExperimentStage::kWarning},
       {"MarkingNoticeAsAcknowledged", MV2ExperimentStage::kDisableWithReEnable},
-      {"PRE_MarkingGlobalNoticeAsAcknowledged", MV2ExperimentStage::kWarning},
       {"MarkingGlobalNoticeAsAcknowledged",
        MV2ExperimentStage::kDisableWithReEnable},
-      {"PRE_PRE_ExtensionsAreReEnabledIfExperimentDisabled",
-       MV2ExperimentStage::kWarning},
       {"PRE_ExtensionsAreReEnabledIfExperimentDisabled",
        MV2ExperimentStage::kDisableWithReEnable},
-      {"ExtensionsAreReEnabledIfExperimentDisabled",
-       MV2ExperimentStage::kWarning},
       {"ExternalExtensionsCanBeInstalledButAreAlsoDisabled",
        MV2ExperimentStage::kDisableWithReEnable},
       {"UkmIsEmittedForExtensionWhenUninstalled",
@@ -84,10 +81,11 @@ MV2ExperimentStage GetExperimentStageForTest(std::string_view test_name) {
       {"UkmIsNotEmittedForOtherUninstallations",
        MV2ExperimentStage::kDisableWithReEnable},
       {"PRE_MV2ExtensionsAreNotDisabledIfLegacyExtensionSwitchIsApplied",
-       MV2ExperimentStage::kWarning},
+       MV2ExperimentStage::kDisableWithReEnable},
       {"MV2ExtensionsAreNotDisabledIfLegacyExtensionSwitchIsApplied",
        MV2ExperimentStage::kUnsupported},
-      {"PRE_PRE_FlowFromWarningToUnsupported", MV2ExperimentStage::kWarning},
+      {"PRE_PRE_FlowFromWarningToUnsupported",
+       MV2ExperimentStage::kDisableWithReEnable},
       {"PRE_FlowFromWarningToUnsupported",
        MV2ExperimentStage::kDisableWithReEnable},
       {"FlowFromWarningToUnsupported", MV2ExperimentStage::kUnsupported},
@@ -127,23 +125,13 @@ class ManifestV2ExperimentManagerBrowserTest : public ExtensionBrowserTest {
         testing::UnitTest::GetInstance()->current_test_info()->name());
 
     switch (experiment_stage) {
-      case MV2ExperimentStage::kWarning:
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-        break;
       case MV2ExperimentStage::kDisableWithReEnable:
-        enabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         disabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
         break;
       case MV2ExperimentStage::kUnsupported:
         enabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         break;
     }
 
@@ -172,9 +160,21 @@ class ManifestV2ExperimentManagerBrowserTest : public ExtensionBrowserTest {
         base::BindRepeating([](std::string_view) { return true; }));
   }
 
-  // Since this is testing the MV2 deprecation experiments, we don't want to
-  // bypass their disabling for testing.
-  bool ShouldAllowMV2Extensions() override { return false; }
+  // Since this is testing the MV2 deprecation experiments, we probably don't
+  // want to bypass their disabling for testing. There are some exceptions for
+  // pre-tests that explicitly need to do this as part of the setup.
+  bool ShouldAllowMV2Extensions() override {
+    std::vector<std::string> tests_allowing_mv2 = {
+        "PRE_ExtensionsAreDisabledOnStartup",
+        "PRE_PRE_ExtensionsCanBeReEnabledByUsers",
+        "PRE_MV2ExtensionsAreNotDisabledIfLegacyExtensionSwitchIsApplied",
+        "PRE_PRE_FlowFromWarningToUnsupported",
+    };
+    return std::ranges::contains(
+        tests_allowing_mv2,
+        std::string(
+            testing::UnitTest::GetInstance()->current_test_info()->name()));
+  }
 
   void WaitForExtensionSystemReady() {
     base::RunLoop run_loop;
@@ -255,37 +255,13 @@ class ManifestV2ExperimentManagerBrowserTest : public ExtensionBrowserTest {
 };
 
 // A test series to verify MV2 extensions are disabled on startup.
-// Step 1 (Warning Only Stage): Install an MV2 extension.
+// Step 1: Install an MV2 extension.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       PRE_PRE_ExtensionsAreDisabledOnStartup) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
+                       PRE_ExtensionsAreDisabledOnStartup) {
   const Extension* extension = AddMV2Extension("Test MV2 Extension");
   ASSERT_TRUE(extension);
 }
-// Step 2 (Warning Only Stage): Verify the MV2 extension is still enabled after
-// restarting the browser. Since this is still a PRE_ stage, the disabling
-// experiment isn't active, and MV2 extensions should be unaffected.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       PRE_ExtensionsAreDisabledOnStartup) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  const Extension* extension = GetExtensionByName(
-      "Test MV2 Extension", extension_registry()->enabled_extensions());
-
-  ASSERT_TRUE(extension);
-  const ExtensionId extension_id = extension->id();
-  EXPECT_TRUE(
-      extension_registry()->enabled_extensions().Contains(extension_id));
-
-  EXPECT_TRUE(extension_prefs()->GetDisableReasons(extension_id).empty());
-
-  histogram_tester().ExpectTotalCount(
-      "Extensions.MV2Deprecation.MV2ExtensionState.Internal", 0);
-}
-// Step 3 (Disable Stage): Verify the extension is disabled. Now the disabling
+// Step 2 (Disable Stage): Verify the extension is disabled. Now the disabling
 // experiment is active, and any old MV2 extensions are disabled.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
                        ExtensionsAreDisabledOnStartup) {
@@ -320,16 +296,16 @@ IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
 
 // A test series to verify extensions that are re-enabled by the user do not
 // get re-disabled on subsequent starts.
-// Step 1 (Warning Only Stage): Install an MV2 extension.
+// Step 1: Install an MV2 extension.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
                        PRE_PRE_ExtensionsCanBeReEnabledByUsers) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
+  EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
+            GetActiveExperimentStage());
 
   const Extension* extension = AddMV2Extension("Test MV2 Extension");
   ASSERT_TRUE(extension);
 }
-// Step 2 (Disable Stage): The extension will be disabled by the experiment.
-// Re-enable the extension.
+// Step 2: The extension will be disabled.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
                        PRE_ExtensionsCanBeReEnabledByUsers) {
   EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
@@ -366,8 +342,8 @@ IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
       static_cast<int64_t>(ManifestV2ExperimentManager::
                                ExtensionMV2DeprecationAction::kReEnabled));
 }
-// Step 3 (Disable Stage): The extension should still be enabled on a subsequent
-// start since the user explicitly chose to re-enable it.
+// Step 3: The extension should still be enabled on a subsequent start since
+// the user explicitly chose to re-enable it.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
                        ExtensionsCanBeReEnabledByUsers) {
   EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
@@ -450,140 +426,6 @@ IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
   EXPECT_FALSE(WasExtensionReEnabledByUser(extension_id));
 }
 
-// Tests that the MV2 deprecation notice for an extension is only acknowledged
-// for the current stage.
-// Step 1 (Warning Stage): Mark an extension's notice as acknowledged on this
-// stage.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       PRE_MarkingNoticeAsAcknowledged) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  // Add an extension and verify it's notice is not marked as acknowledged on
-  // this stage.
-  const Extension* extension = AddMV2Extension("Test MV2 Extension");
-  ASSERT_TRUE(extension);
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNotice(extension->id()));
-
-  // Mark the notice as acknowledged for this stage. Verify it's acknowledged.
-  experiment_manager()->MarkNoticeAsAcknowledged(extension->id());
-  EXPECT_TRUE(experiment_manager()->DidUserAcknowledgeNotice(extension->id()));
-}
-// Step 2 (Disable Stage): Verify extension's notice is not acknowledged on this
-// stage. Mark notice as acknowledged on this stage.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       MarkingNoticeAsAcknowledged) {
-  EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
-            GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  // Verify extension's notice is not marked as acknowledged on this stage, even
-  // if it was acknowledged on the previous stage.
-  const Extension* extension = GetExtensionByName(
-      "Test MV2 Extension", extension_registry()->disabled_extensions());
-  ASSERT_TRUE(extension);
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNotice(extension->id()));
-
-  // Mark the notice as acknowledged for this stage. Verify it's acknowledged.
-  experiment_manager()->MarkNoticeAsAcknowledged(extension->id());
-  EXPECT_TRUE(experiment_manager()->DidUserAcknowledgeNotice(extension->id()));
-}
-
-// Tests that the MV2 deprecation global notice is only acknowledged for the
-// current stage.
-// Step 1 (Warning Stage): Mark the global notice as acknowledged
-// on this stage.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       PRE_MarkingGlobalNoticeAsAcknowledged) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  // Add an extension that should make the MV2 deprecation notice visible.
-  // Verify global notice is not marked as acknowledged on this stage.
-  const Extension* extension = AddMV2Extension("Test MV2 Extension");
-  ASSERT_TRUE(extension);
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNoticeGlobally());
-
-  // Mark the global notice as acknowledged for this stage. Verify it's
-  // acknowledged.
-  experiment_manager()->MarkNoticeAsAcknowledgedGlobally();
-  EXPECT_TRUE(experiment_manager()->DidUserAcknowledgeNoticeGlobally());
-}
-// Step 2 (Disable Stage): Verify global notice is not acknowledged on this
-// stage. Mark notice as acknowledged on this stage.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       MarkingGlobalNoticeAsAcknowledged) {
-  EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
-            GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  // Verify global notice is not marked as acknowledged on this stage, even if
-  // it was acknowledged on the previous stage.
-  const Extension* extension = GetExtensionByName(
-      "Test MV2 Extension", extension_registry()->disabled_extensions());
-  ASSERT_TRUE(extension);
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNoticeGlobally());
-
-  // Mark the global notice as acknowledged for this stage. Verify it's
-  // acknowledged.
-  experiment_manager()->MarkNoticeAsAcknowledgedGlobally();
-  EXPECT_TRUE(experiment_manager()->DidUserAcknowledgeNoticeGlobally());
-}
-
-// Tests that if a user moves from a later experiment stage (disable with
-// re-enable) to an earlier one (warning), any disabled extensions will be
-// automatically re-enabled.
-// First stage: install an MV2 extension.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       PRE_PRE_ExtensionsAreReEnabledIfExperimentDisabled) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
-  const Extension* extension = AddMV2Extension("Test MV2 Extension");
-  ASSERT_TRUE(extension);
-}
-// Second stage: MV2 deprecation experiment takes effect; extension is disabled.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       PRE_ExtensionsAreReEnabledIfExperimentDisabled) {
-  EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
-            GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  const Extension* extension = GetExtensionByName(
-      "Test MV2 Extension", extension_registry()->disabled_extensions());
-  ASSERT_TRUE(extension);
-  const ExtensionId extension_id = extension->id();
-  EXPECT_THAT(extension_prefs()->GetDisableReasons(extension_id),
-              testing::UnorderedElementsAre(
-                  disable_reason::DISABLE_UNSUPPORTED_MANIFEST_VERSION));
-}
-// Third stage: Move the user back to the warning stage. The extension should be
-// re-enabled.
-IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
-                       ExtensionsAreReEnabledIfExperimentDisabled) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
-  WaitForExtensionSystemReady();
-
-  const Extension* extension = GetExtensionByName(
-      "Test MV2 Extension", extension_registry()->enabled_extensions());
-  ASSERT_TRUE(extension);
-  const ExtensionId extension_id = extension->id();
-
-  EXPECT_TRUE(extension_prefs()->GetDisableReasons(extension_id).empty());
-  // The user didn't re-enable the extension, so it shouldn't be marked as such.
-  EXPECT_FALSE(WasExtensionReEnabledByUser(extension_id));
-
-  // Since the user is no longer in the disable phase, no metrics should be
-  // reported.
-  histogram_tester().ExpectTotalCount(
-      "Extensions.MV2Deprecation.MV2ExtensionState.Internal", 0);
-}
-
 // Tests that externally-installed extensions are allowed to be installed, but
 // will still be disabled by the MV2 experiments.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
@@ -698,9 +540,10 @@ IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
 }
 
 // Tests the flow from the "warning" phase to the "unsupported" phase.
+// Since the "warning" phase is removed, this imitates it by just adding two
+// MV2 extensions.
 IN_PROC_BROWSER_TEST_F(ManifestV2ExperimentManagerBrowserTest,
                        PRE_PRE_FlowFromWarningToUnsupported) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
   // Install two MV2 extensions.
   const Extension* extension1 = AddMV2Extension("Test MV2 Extension 1");
   ASSERT_TRUE(extension1);
@@ -911,8 +754,6 @@ class ManifestV2ExperimentWithLegacyExtensionSupportTest
 IN_PROC_BROWSER_TEST_F(
     ManifestV2ExperimentWithLegacyExtensionSupportTest,
     PRE_MV2ExtensionsAreNotDisabledIfLegacyExtensionSwitchIsApplied) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning, GetActiveExperimentStage());
-
   // Load two extensions: a packed extension and an unpacked extension.
   const Extension* packed_extension =
       AddMV2Extension("Test Packed MV2 Extension");
diff --git a/chrome/browser/extensions/manifest_v2_experiment_manager_unittest.cc b/chrome/browser/extensions/manifest_v2_experiment_manager_unittest.cc
index b809a1f608501..dcd66bf430797 100644
--- a/chrome/browser/extensions/manifest_v2_experiment_manager_unittest.cc
+++ b/chrome/browser/extensions/manifest_v2_experiment_manager_unittest.cc
@@ -81,18 +81,6 @@ ManifestV2ExperimentManagerUnitTestBase::
   feature_list_.InitWithFeatures(enabled_features, disabled_features);
 }
 
-// Test suite for cases where the user is in the "warning" experiment phase.
-class ManifestV2ExperimentManagerWarningUnitTest
-    : public ManifestV2ExperimentManagerUnitTestBase {
- public:
-  ManifestV2ExperimentManagerWarningUnitTest()
-      : ManifestV2ExperimentManagerUnitTestBase(
-            {},
-            {extensions_features::kExtensionManifestV2Disabled,
-             extensions_features::kExtensionManifestV2Unsupported}) {}
-  ~ManifestV2ExperimentManagerWarningUnitTest() override = default;
-};
-
 // Test suite for cases where the user is in the "disable with re-enable"
 // experiment phase.
 class ManifestV2ExperimentManagerDisableWithReEnableUnitTest
@@ -100,24 +88,11 @@ class ManifestV2ExperimentManagerDisableWithReEnableUnitTest
  public:
   ManifestV2ExperimentManagerDisableWithReEnableUnitTest()
       : ManifestV2ExperimentManagerUnitTestBase(
-            {extensions_features::kExtensionManifestV2Disabled},
+            {},
             {extensions_features::kExtensionManifestV2Unsupported}) {}
   ~ManifestV2ExperimentManagerDisableWithReEnableUnitTest() override = default;
 };
 
-// Test suite for cases where the user is in the "disable with re-enable"
-// experiment phase *and* the warning experiment is still active.
-class ManifestV2ExperimentManagerDisableWithReEnableAndWarningUnitTest
-    : public ManifestV2ExperimentManagerUnitTestBase {
- public:
-  ManifestV2ExperimentManagerDisableWithReEnableAndWarningUnitTest()
-      : ManifestV2ExperimentManagerUnitTestBase(
-            {extensions_features::kExtensionManifestV2Disabled},
-            {extensions_features::kExtensionManifestV2Unsupported}) {}
-  ~ManifestV2ExperimentManagerDisableWithReEnableAndWarningUnitTest() override =
-      default;
-};
-
 // Test suite for cases where the user is in the "unsupported" experiment phase.
 class ManifestV2ExperimentManagerUnsupportedUnitTest
     : public ManifestV2ExperimentManagerUnitTestBase {
@@ -129,89 +104,6 @@ class ManifestV2ExperimentManagerUnsupportedUnitTest
   ~ManifestV2ExperimentManagerUnsupportedUnitTest() override = default;
 };
 
-// Tests that the experiment stage is properly set when the manifest V2
-// deprecation warning experiment is enabled.
-TEST_F(ManifestV2ExperimentManagerWarningUnitTest,
-       ExperimentStageIsSetToWarning) {
-  EXPECT_EQ(MV2ExperimentStage::kWarning,
-            experiment_manager()->GetCurrentExperimentStage());
-}
-
-// Sanity check that MV2 extensions are considered affected when the
-// experiment is enabled. The "is affected" logic is much more heavily tested
-// in mv2_deprecation_impact_checker_unittest.cc.
-TEST_F(ManifestV2ExperimentManagerWarningUnitTest, MV2ExtensionsAreAffected) {
-  struct {
-    mojom::ManifestLocation manifest_location;
-    const char* name;
-  } test_cases[] = {
-      {mojom::ManifestLocation::kInternal, "internal"},
-      {mojom::ManifestLocation::kExternalPref, "external pref"},
-      {mojom::ManifestLocation::kExternalRegistry, "external registry"},
-  };
-
-  for (const auto& test_case : test_cases) {
-    SCOPED_TRACE(test_case.name);
-    scoped_refptr<const Extension> mv2_extension =
-        ExtensionBuilder(test_case.name)
-            .SetManifestVersion(2)
-            .SetLocation(test_case.manifest_location)
-            .Build();
-    EXPECT_TRUE(experiment_manager()->IsExtensionAffected(*mv2_extension));
-    // Even though the MV2 extension is affected by the experiment, it should
-    // *not* be blocked from installation or enablement in the warning phase.
-    EXPECT_FALSE(experiment_manager()->ShouldBlockExtensionInstallation(
-        mv2_extension->id(), mv2_extension->manifest_version(),
-        mv2_extension->GetType(), mv2_extension->location(),
-        mv2_extension->hashed_id()));
-    EXPECT_FALSE(
-        experiment_manager()->ShouldBlockExtensionEnable(*mv2_extension));
-
-    // Modern extensions are not affected by the experiment.
-    scoped_refptr<const Extension> extension =
-        ExtensionBuilder(test_case.name)
-            .SetLocation(test_case.manifest_location)
-            .Build();
-    EXPECT_FALSE(experiment_manager()->IsExtensionAffected(*extension));
-    EXPECT_FALSE(experiment_manager()->ShouldBlockExtensionInstallation(
-        extension->id(), extension->manifest_version(), extension->GetType(),
-        extension->location(), extension->hashed_id()));
-    EXPECT_FALSE(experiment_manager()->ShouldBlockExtensionEnable(*extension));
-  }
-}
-
-TEST_F(ManifestV2ExperimentManagerWarningUnitTest,
-       MarkingNoticeAsAcknowledged) {
-  scoped_refptr<const Extension> ext1 =
-      ExtensionBuilder("one")
-          .SetManifestVersion(2)
-          .SetLocation(mojom::ManifestLocation::kInternal)
-          .Build();
-  scoped_refptr<const Extension> ext2 =
-      ExtensionBuilder("two")
-          .SetManifestVersion(2)
-          .SetLocation(mojom::ManifestLocation::kInternal)
-          .Build();
-
-  registrar()->AddExtension(ext1.get());
-  registrar()->AddExtension(ext2.get());
-
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNotice(ext1->id()));
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNotice(ext2->id()));
-
-  experiment_manager()->MarkNoticeAsAcknowledged(ext1->id());
-
-  EXPECT_TRUE(experiment_manager()->DidUserAcknowledgeNotice(ext1->id()));
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNotice(ext2->id()));
-}
-
-TEST_F(ManifestV2ExperimentManagerWarningUnitTest,
-       MarkingGlobalNoticeAsAcknowledged) {
-  EXPECT_FALSE(experiment_manager()->DidUserAcknowledgeNoticeGlobally());
-  experiment_manager()->MarkNoticeAsAcknowledgedGlobally();
-  EXPECT_TRUE(experiment_manager()->DidUserAcknowledgeNoticeGlobally());
-}
-
 // Tests that the experiment phase is properly set for a user in the
 // "disable with re-enable" experiment phase.
 TEST_F(ManifestV2ExperimentManagerDisableWithReEnableUnitTest,
@@ -220,15 +112,6 @@ TEST_F(ManifestV2ExperimentManagerDisableWithReEnableUnitTest,
             experiment_manager()->GetCurrentExperimentStage());
 }
 
-// Tests that the experiment phase is properly set for a user in the
-// "disable with re-enable" experiment phase if the "warning" experiment is
-// still active. That is, verifies that the "latest" stage takes precedence.
-TEST_F(ManifestV2ExperimentManagerDisableWithReEnableAndWarningUnitTest,
-       ExperimentStageIsSetToDisableWithReEnable) {
-  EXPECT_EQ(MV2ExperimentStage::kDisableWithReEnable,
-            experiment_manager()->GetCurrentExperimentStage());
-}
-
 // Sanity check that MV2 extensions are considered affected when the
 // experiment is enabled in the "disable with re-enable" phase. The
 // "is affected" logic is much more heavily tested
diff --git a/chrome/browser/extensions/mv2_deprecation_impact_checker_unittest.cc b/chrome/browser/extensions/mv2_deprecation_impact_checker_unittest.cc
index 8ad8b73e6445d..37156bb351781 100644
--- a/chrome/browser/extensions/mv2_deprecation_impact_checker_unittest.cc
+++ b/chrome/browser/extensions/mv2_deprecation_impact_checker_unittest.cc
@@ -67,9 +67,6 @@ std::string DescribeTestVariant(const TestVariant& test_variant) {
   description += "And";
 
   switch (experiment_stage) {
-    case MV2ExperimentStage::kWarning:
-      description += "WarningExperiment";
-      break;
     case MV2ExperimentStage::kDisableWithReEnable:
       description += "DisableExperiment";
       break;
@@ -233,8 +230,7 @@ INSTANTIATE_TEST_SUITE_P(
     ,
     MV2DeprecationImpactCheckerUnitTest,
     testing::Combine(
-        testing::Values(MV2ExperimentStage::kWarning,
-                        MV2ExperimentStage::kDisableWithReEnable,
+        testing::Values(MV2ExperimentStage::kDisableWithReEnable,
                         MV2ExperimentStage::kUnsupported),
         testing::Values(MV2PolicyLevel::kUnset,
                         MV2PolicyLevel::kAllowed,
@@ -248,8 +244,7 @@ INSTANTIATE_TEST_SUITE_P(
     ,
     MV2DeprecationImpactCheckerUnitTestWithAllowlist,
     testing::Combine(
-        testing::Values(MV2ExperimentStage::kWarning,
-                        MV2ExperimentStage::kDisableWithReEnable,
+        testing::Values(MV2ExperimentStage::kDisableWithReEnable,
                         MV2ExperimentStage::kUnsupported),
         testing::Values(MV2PolicyLevel::kUnset,
                         MV2PolicyLevel::kAllowed,
diff --git a/chrome/browser/extensions/scoped_test_mv2_enabler_browsertest.cc b/chrome/browser/extensions/scoped_test_mv2_enabler_browsertest.cc
index bce7c9d5704d2..9b700ce3e808b 100644
--- a/chrome/browser/extensions/scoped_test_mv2_enabler_browsertest.cc
+++ b/chrome/browser/extensions/scoped_test_mv2_enabler_browsertest.cc
@@ -5,7 +5,6 @@
 #include "chrome/browser/extensions/scoped_test_mv2_enabler.h"
 
 #include "base/one_shot_event.h"
-#include "base/test/scoped_feature_list.h"
 #include "chrome/browser/extensions/extension_browsertest.h"
 #include "chrome/browser/profiles/profile.h"
 #include "content/public/test/browser_test.h"
@@ -14,21 +13,16 @@
 #include "extensions/browser/extension_system.h"
 #include "extensions/browser/manifest_v2_experiment_manager.h"
 #include "extensions/common/extension.h"
-#include "extensions/common/extension_features.h"
 #include "extensions/test/test_extension_dir.h"
 
 namespace extensions {
 
 class ScopedTestMV2EnablerBrowserTest : public ExtensionBrowserTest {
  public:
-  ScopedTestMV2EnablerBrowserTest() {
-    feature_list_.InitAndEnableFeature(
-        extensions_features::kExtensionManifestV2Disabled);
-  }
+  ScopedTestMV2EnablerBrowserTest() = default;
   ~ScopedTestMV2EnablerBrowserTest() override = default;
 
  private:
-  base::test::ScopedFeatureList feature_list_;
   ScopedTestMV2Enabler mv2_enabler_;
 };
 
diff --git a/chrome/browser/flag-metadata.json b/chrome/browser/flag-metadata.json
index 1d00456026737..25f1a48581d78 100644
--- a/chrome/browser/flag-metadata.json
+++ b/chrome/browser/flag-metadata.json
@@ -4528,11 +4528,6 @@
     "owners": [ "richche@google.com" ],
     "expiry_milestone": 143
   },
-  {
-    "name": "extension-manifest-v2-deprecation-disabled",
-    "owners": [ "emiliapaz@chromium.org" ],
-    "expiry_milestone": 137
-  },
   {
     "name": "extension-manifest-v2-deprecation-unsupported",
     "owners": [ "emiliapaz@chromium.org" ],
diff --git a/chrome/browser/flag_descriptions.h b/chrome/browser/flag_descriptions.h
index 325b6a0f2bd29..cff4dd74a47fd 100644
--- a/chrome/browser/flag_descriptions.h
+++ b/chrome/browser/flag_descriptions.h
@@ -2305,11 +2305,6 @@ inline constexpr char kIPHExtensionsRequestAccessButtonFeatureDescription[] =
     "Enables In-Product-Help that appears when at least one extension is "
     "requesting access to the current page. This feature is gated by "
     "extensions-menu-access-control.";
-inline constexpr char kExtensionManifestV2DeprecationDisabledName[] =
-    "Extension Manifest V2 Deprecation Disabled Stage";
-inline constexpr char kExtensionManifestV2DeprecationDisabledDescription[] =
-    "Displays a warning that affected MV2 extensions were turned off due to "
-    "the Manifest V2 deprecation.";
 inline constexpr char kExtensionManifestV2DeprecationUnsupportedName[] =
     "Extension Manifest V2 Deprecation Unsupported Stage";
 inline constexpr char kExtensionManifestV2DeprecationUnsupportedDescription[] =
diff --git a/chrome/browser/ui/extensions/mv2_disabled_dialog_controller.cc b/chrome/browser/ui/extensions/mv2_disabled_dialog_controller.cc
index b26dc8cf4503b..b18abdf6e1f13 100644
--- a/chrome/browser/ui/extensions/mv2_disabled_dialog_controller.cc
+++ b/chrome/browser/ui/extensions/mv2_disabled_dialog_controller.cc
@@ -49,10 +49,6 @@ constexpr PrefMap kMV2DeprecationUnsupportedDisabledDialogAcknowledgedPref = {
 const PrefMap& GetDisabledDialogAcknowledgedPref(
     MV2ExperimentStage experiment_stage) {
   switch (experiment_stage) {
-    case MV2ExperimentStage::kWarning:
-      // There is no disabled dialog for this stage, thus extension cannot be
-      // acknowledged.
-      NOTREACHED();
     case MV2ExperimentStage::kDisableWithReEnable:
       return kMV2DeprecationDisabledDialogAcknowledgedPref;
     case MV2ExperimentStage::kUnsupported:
diff --git a/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_keep_dialog_interactive_uitest.cc b/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_keep_dialog_interactive_uitest.cc
index 2da322610b9e3..ef5db4ecab9c7 100644
--- a/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_keep_dialog_interactive_uitest.cc
+++ b/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_keep_dialog_interactive_uitest.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include "base/functional/callback_helpers.h"
-#include "base/test/scoped_feature_list.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_element_identifiers.h"
@@ -15,15 +14,11 @@
 #include "extensions/browser/extension_system.h"
 #include "extensions/common/extension.h"
 #include "extensions/common/extension_builder.h"
-#include "extensions/common/extension_features.h"
 #include "extensions/common/mojom/manifest.mojom.h"
 
 class Mv2DeprecationKeepDialogInteractiveTest : public InteractiveBrowserTest {
  public:
-  Mv2DeprecationKeepDialogInteractiveTest() {
-    scoped_feature_list_.InitAndEnableFeature(
-        extensions_features::kExtensionManifestV2Disabled);
-  }
+  Mv2DeprecationKeepDialogInteractiveTest() = default;
   ~Mv2DeprecationKeepDialogInteractiveTest() override = default;
   Mv2DeprecationKeepDialogInteractiveTest(
       const Mv2DeprecationKeepDialogInteractiveTest&) = delete;
@@ -51,9 +46,6 @@ class Mv2DeprecationKeepDialogInteractiveTest : public InteractiveBrowserTest {
   extensions::ExtensionRegistrar* extension_registrar() {
     return extensions::ExtensionRegistrar::Get(browser()->profile());
   }
-
- private:
-  base::test::ScopedFeatureList scoped_feature_list_;
 };
 
 IN_PROC_BROWSER_TEST_F(Mv2DeprecationKeepDialogInteractiveTest, ShowDialog) {
diff --git a/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_reenable_dialog_browsertest.cc b/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_reenable_dialog_browsertest.cc
index 57a97f2418e5b..25eb3f5b15e77 100644
--- a/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_reenable_dialog_browsertest.cc
+++ b/chrome/browser/ui/views/extensions/dialogs/mv2_deprecation_reenable_dialog_browsertest.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include "base/functional/callback_helpers.h"
-#include "base/test/scoped_feature_list.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_window.h"
@@ -14,16 +13,12 @@
 #include "extensions/browser/extension_system.h"
 #include "extensions/common/extension.h"
 #include "extensions/common/extension_builder.h"
-#include "extensions/common/extension_features.h"
 #include "extensions/common/mojom/manifest.mojom.h"
 
 class Mv2DeprecationReEnableDialogBrowserTest
     : public ExtensionsDialogBrowserTest {
  public:
-  Mv2DeprecationReEnableDialogBrowserTest() {
-    scoped_feature_list_.InitAndEnableFeature(
-        extensions_features::kExtensionManifestV2Disabled);
-  }
+  Mv2DeprecationReEnableDialogBrowserTest() = default;
   ~Mv2DeprecationReEnableDialogBrowserTest() override = default;
   Mv2DeprecationReEnableDialogBrowserTest(
       const Mv2DeprecationReEnableDialogBrowserTest&) = delete;
@@ -60,9 +55,6 @@ class Mv2DeprecationReEnableDialogBrowserTest
         browser()->window()->GetNativeWindow(), extension->id(),
         extension->name(), /*done_callback=*/base::DoNothing());
   }
-
- private:
-  base::test::ScopedFeatureList scoped_feature_list_;
 };
 
 IN_PROC_BROWSER_TEST_F(Mv2DeprecationReEnableDialogBrowserTest,
diff --git a/chrome/browser/ui/views/extensions/dialogs/mv2_disabled_dialog_controller_interactive_uitest.cc b/chrome/browser/ui/views/extensions/dialogs/mv2_disabled_dialog_controller_interactive_uitest.cc
index 275c85c488a37..0ded2f13b4e84 100644
--- a/chrome/browser/ui/views/extensions/dialogs/mv2_disabled_dialog_controller_interactive_uitest.cc
+++ b/chrome/browser/ui/views/extensions/dialogs/mv2_disabled_dialog_controller_interactive_uitest.cc
@@ -183,18 +183,13 @@ class Mv2DisabledDialogControllerInteractiveUITest
     std::vector<base::test::FeatureRef> disabled_features;
 
     switch (experiment_stage) {
-      case MV2ExperimentStage::kWarning:
       case MV2ExperimentStage::kDisableWithReEnable:
-        enabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         disabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
         break;
       case MV2ExperimentStage::kUnsupported:
         enabled_features.push_back(
             extensions_features::kExtensionManifestV2Unsupported);
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Disabled);
         break;
     }
     feature_list_.InitWithFeatures(enabled_features, disabled_features);
@@ -228,8 +223,6 @@ INSTANTIATE_TEST_SUITE_P(
                       MV2ExperimentStage::kUnsupported),
     [](const testing::TestParamInfo<MV2ExperimentStage>& info) {
       switch (info.param) {
-        case MV2ExperimentStage::kWarning:
-          NOTREACHED();
         case MV2ExperimentStage::kDisableWithReEnable:
           return "DisableExperiment";
         case MV2ExperimentStage::kUnsupported:
diff --git a/extensions/browser/extension_prefs.cc b/extensions/browser/extension_prefs.cc
index 71e6d4d37e7c9..2a88070f7b411 100644
--- a/extensions/browser/extension_prefs.cc
+++ b/extensions/browser/extension_prefs.cc
@@ -2667,6 +2667,9 @@ void ExtensionPrefs::MigrateObsoleteExtensionPrefs() {
 
       // Added 2025-08.
       "state",
+
+      // Added 2026-05.
+      "mv2_deprecation_warning_ack",
   };
 
   for (auto key_value : extensions_dictionary) {
diff --git a/extensions/browser/manifest_v2_experiment_manager.cc b/extensions/browser/manifest_v2_experiment_manager.cc
index 4fa8e5ebed2e7..e210ea8310ae7 100644
--- a/extensions/browser/manifest_v2_experiment_manager.cc
+++ b/extensions/browser/manifest_v2_experiment_manager.cc
@@ -64,12 +64,6 @@ const char* GetHistogramManifestLocation(mojom::ManifestLocation location) {
   }
 }
 
-// Stores the bit for whether the user has acknowledged the MV2 deprecation
-// notice for a given extension in the warning stage.
-constexpr PrefMap kMV2DeprecationExtensionWarningAcknowledgedPref = {
-    "mv2_deprecation_warning_ack", PrefType::kBool,
-    PrefScope::kExtensionSpecific};
-
 // Stores the bit for whether the user has acknowledged the MV2 deprecation
 // notice for a given extension in the disabled stage.
 constexpr PrefMap kMV2DeprecationExtensionDisabledAcknowledgedPref = {
@@ -157,20 +151,13 @@ MV2ExperimentStage CalculateCurrentExperimentStage() {
     return MV2ExperimentStage::kUnsupported;
   }
 
-  if (base::FeatureList::IsEnabled(
-          extensions_features::kExtensionManifestV2Disabled)) {
-    return MV2ExperimentStage::kDisableWithReEnable;
-  }
-
-  return MV2ExperimentStage::kWarning;
+  return MV2ExperimentStage::kDisableWithReEnable;
 }
 
 // Returns the pref that stores whether the user has acknowledged the MV2
 // deprecation notice for a given extension in `experiment_stage`.
 PrefMap GetExtensionAcknowledgedPrefFor(MV2ExperimentStage experiment_stage) {
   switch (experiment_stage) {
-    case MV2ExperimentStage::kWarning:
-      return kMV2DeprecationExtensionWarningAcknowledgedPref;
     case MV2ExperimentStage::kDisableWithReEnable:
       return kMV2DeprecationExtensionDisabledAcknowledgedPref;
     case MV2ExperimentStage::kUnsupported:
@@ -183,8 +170,6 @@ PrefMap GetExtensionAcknowledgedPrefFor(MV2ExperimentStage experiment_stage) {
 PrefMap GetGlobalNoticeAcknowledgedPrefFor(
     MV2ExperimentStage experiment_stage) {
   switch (experiment_stage) {
-    case MV2ExperimentStage::kWarning:
-      return kMV2DeprecationWarningAcknowledgedGloballyPref;
     case MV2ExperimentStage::kDisableWithReEnable:
       return kMV2DeprecationDisabledAcknowledgedGloballyPref;
     case MV2ExperimentStage::kUnsupported:
@@ -200,13 +185,7 @@ bool ShouldDisableLegacyExtensions(MV2ExperimentStage stage) {
     return false;
   }
 
-  switch (stage) {
-    case MV2ExperimentStage::kWarning:
-      return false;
-    case MV2ExperimentStage::kDisableWithReEnable:
-    case MV2ExperimentStage::kUnsupported:
-      return true;
-  }
+  return true;
 }
 
 // Returns true if the given `stage` is one in which extension enablement should
@@ -221,7 +200,6 @@ bool ShouldBlockLegacyExtensionEnableForStage(MV2ExperimentStage stage) {
   // We only block extension enablement in the `kUnsupported` phase.
   // (We use a switch just to ensure compile errors if we ever add a new phase.)
   switch (stage) {
-    case MV2ExperimentStage::kWarning:
     case MV2ExperimentStage::kDisableWithReEnable:
       return false;
     case MV2ExperimentStage::kUnsupported:
@@ -251,7 +229,6 @@ bool ShouldBlockUnpackedExtensions(MV2ExperimentStage stage) {
 // given experiment `stage`.
 bool UserCanReEnableExtensionsForStage(MV2ExperimentStage stage) {
   switch (stage) {
-    case MV2ExperimentStage::kWarning:
     case MV2ExperimentStage::kDisableWithReEnable:
       return true;
     case MV2ExperimentStage::kUnsupported:
diff --git a/extensions/browser/mv2_experiment_stage.h b/extensions/browser/mv2_experiment_stage.h
index e9c456302fbe5..39ffe5b3dabac 100644
--- a/extensions/browser/mv2_experiment_stage.h
+++ b/extensions/browser/mv2_experiment_stage.h
@@ -14,12 +14,9 @@ namespace extensions {
 // The current stage of the Manifest V2 deprecation. Note that for all stages,
 // these only refer to affected extensions.
 enum class MV2ExperimentStage {
-  // The user is warned that Manifest V2 extensions are deprecated and will
-  // soon be unsupported.
-  // This is the lowest possible stage (launched by default).
-  kWarning,
   // Unsupported Manifest V2 extensions will be disabled. The user is allowed
   // to re-enable them at any point.
+  // This is the lowest possible stage (launched by default).
   kDisableWithReEnable,
   // Manifest V2 extensions are no longer supported. The user is not able to
   // re-enable them.
diff --git a/extensions/common/extension_features.cc b/extensions/common/extension_features.cc
index 11ab8c28e93c0..6346d1df3f7d3 100644
--- a/extensions/common/extension_features.cc
+++ b/extensions/common/extension_features.cc
@@ -116,8 +116,6 @@ BASE_FEATURE(kExtensionManifestV2Unsupported, base::FEATURE_ENABLED_BY_DEFAULT);
 BASE_FEATURE(kExtensionManifestV2ExceptionList,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kExtensionManifestV2Disabled, base::FEATURE_ENABLED_BY_DEFAULT);
-
 BASE_FEATURE(kExtensionsBackgroundCompilation,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
diff --git a/extensions/common/extension_features.h b/extensions/common/extension_features.h
index 833b762d73232..7b67500f4773f 100644
--- a/extensions/common/extension_features.h
+++ b/extensions/common/extension_features.h
@@ -152,10 +152,6 @@ BASE_DECLARE_FEATURE(kExtensionLocalizationGuid);
 // A replacement key for declaring icons, in addition to supporting dark mode.
 BASE_DECLARE_FEATURE(kExtensionIconVariants);
 
-// Controls disabling affected MV2 extensions that are no longer supported.
-// Users can re-enable these extensions.
-BASE_DECLARE_FEATURE(kExtensionManifestV2Disabled);
-
 // Controls fully removing support for user-installed MV2 extensions.
 // Users may no longer re-enable these extensions. Enterprises may still
 // override this.
